docs/src/content/docs/id/guides/components.mdx

```diff
@@ -163,6 +164,198 @@ import { LinkCard } from '@astrojs/starlight/components';
+Lihat panduan [“Penulisan Konten di Markdown”](/id/guides/authoring-content/#asides) untuk detail dari siktaks khusus.
+
+### Code
+
+Menggunakan komponen `<Code>` untuk merender _syntax highlighted code_ ketika menggunakan [Markdown code block](/id/guides/authoring-content/#code-blocks) tidak memungkinkan, contohnya, untuk merender data dari sumber eksternal seperti file, database, atau API.
+
+Lihat [Expressive Code dokumentasi “Komponen Kode”](https://expressive-code.com/key-features/code-component/) untuk detail lengkap tentang dukungan props `<Code>`.
+
+```mdx
+# src/content/docs/example.mdx
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('Ini bisa berasal dari file atau CMS!');`;
+export const fileName = 'example.js';
+export const highlights = ['file', 'CMS'];
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+```
+
+Kode di atas akan menampilkan hasil sebagai berikut di halaman:
+
+import { Code } from '@astrojs/starlight/components';
+
+export const exampleCode = `console.log('Ini bisa berasal dari file atau CMS!');`;
+export const fileName = 'example.js';
+export const highlights = ['file', 'CMS'];
+
+<Code code={exampleCode} lang="js" title={fileName} mark={highlights} />
+
+#### Imported Code
+
+Gunakan [Sufiks impor `?raw` Vite](https://vitejs.dev/guide/assets#importing-asset-as-string) to impor file kode apa pun sebagai string.
+Anda dapat meneruskan string yang diimpor ini ke komponen `<Code>` untuk memasukkannya ke halaman Anda.
+
+```mdx title="src/content/docs/example.mdx" "?raw"
+import { Code } from '@astrojs/starlight/components';
+import importedCode from '/src/env.d.ts?raw';
+
+<Code code={importedCode} lang="ts" title="src/env.d.ts" />
+```
+
+Kode di atas akan menampilkan hasil sebagai berikut di halaman:
+
+import importedCode from '/src/env.d.ts?raw';
+
+<Code code={importedCode} lang="ts" title="src/env.d.ts" />
+
+### File Tree
+
+Gunakan komponen `<FileTree>` untuk menampilkan struktur direktori dengan ikon berkas beserta subdirektori yang dapat diciutkan.
+
+Tentukan struktur file dan direktori Anda dengan [daftar Markdown tidak berurutan](https://www.markdownguide.org/basic-syntax/#unordered-lists) di dalam `<FileTree>`.
+Buat subdirektori menggunakan daftar bersarang atau tambahkan `/` diakhir item untuk merendernya sebagai direktori tanpa konten yang spesifik.
+
+Sintaks berikut dapat digunakan untuk menyesuaikan tampilan pohon file:
+
+The following syntax can be used to customize the appearance of the file tree:
+
+- Sorot file atau direktori dengan membuat namanya tebal, mis. `**README.md**`.
+- Menambah komen ke berkas atau direktori dengan menambah teks setelah nama.
+- Menambah _placeholder_ untuk berkas dan direktori dengan memakai `...` or `…` sebagai nama.
+
+```mdx
+# src/content/docs/example.mdx
+
+import { FileTree } from '@astrojs/starlight/components';
+
+<FileTree>
+
+- astro.config.mjs an **important** file
```

Comment:
Do you think it would make sense to translate the "an important file comment". We did it in other languages so if we can, I think it could help a bit. Let me know what you think.